oci/casext/mediatype/parse.go

```diff
@@ -136,22 +137,66 @@ func CustomJSONParser(v interface{}) ParseFunc {
 	if t.Kind() != reflect.Struct {
 		panic("CustomJSONParser() called with non-struct kind!")
 	}
-	return func(reader io.Reader) (_ interface{}, err error) {
+	return func(rdr io.Reader) (_ interface{}, err error) {
 		ptr := reflect.New(t)
-		if reader != nil {
-			err = json.NewDecoder(reader).Decode(ptr.Interface())
+		if rdr != nil {
+			err = json.NewDecoder(rdr).Decode(ptr.Interface())
```

Comment:
we don't need this hunk, i guess?

Reply:
It was to make it consistent with the other methods (I've always preferred to use `rdr` anyway).

Reply:
ok. might be worth a separate patch for it then.